projects/plugins/jetpack/_inc/client/security/firewall.jsx

```diff
@@ -0,0 +1,305 @@
+
+export const Firewall = class extends Component {
+	/**
+	 * Get options for initial state.
+	 *
+	 * @returns {{jetpack_firewall_ip_list: *, jetpack_firewall_share_data: *}}
+	 */
+	state = {
+		jetpack_firewall_ip_list: this.props.getOptionValue( 'jetpack_firewall_ip_list' ),
+		jetpack_firewall_ip_allow_list: this.props.getOptionValue( 'jetpack_firewall_ip_allow_list' ),
+		jetpack_firewall_ip_block_list: this.props.getOptionValue( 'jetpack_firewall_ip_block_list' ),
+		jetpack_firewall_share_data: this.props.getOptionValue( 'jetpack_firewall_share_data' ),
+	};
+
+	handleIpListToggleChange = () => {
+		this.updateOptions( 'jetpack_firewall_ip_list' );
+	};
+
+	handleShareDataToggleChange = () => {
+		this.updateOptions( 'jetpack_firewall_share_data' );
+	};
+
+	handleIpListChange = event => {
+		const {
+			target: { name, value },
+		} = event;
+
+		this.setState( {
+			[ name ]: value,
+		} );
+
+		this.props.onOptionChange( event );
+	};
+
+	/**
+	 * Update state so toggles are updated.
+	 *
+	 * @param {string} optionName - The slug of the option to update
+	 */
+	updateOptions = optionName => {
+		this.setState(
+			{
+				[ optionName ]: ! this.state[ optionName ],
+			},
+			this.props.updateFormStateModuleOption( 'firewall', optionName )
+		);
+	};
+
+	render() {
+		if ( ! this.props.uiEnabled ) {
+			return <QueryWafSettings />;
+		}
+
+		const isFirewallActive = this.props.getOptionValue( 'firewall' ),
+			unavailableInOfflineMode = this.props.isUnavailableInOfflineMode( 'firewall' );
+
+		const moduleHeader = (
+			<div className="firewall__header">
+				<span>{ _x( 'Firewall', 'Settings header', 'jetpack' ) }</span>
+				<span className="firewall__header__badge">
+					{ _x( 'Beta', 'Settings header badge', 'jetpack' ) }
+				</span>
+			</div>
+		);
+
+		const ipListSettings = (
+			<div className="firewall__settings__toggle-setting">
+				<CompactFormToggle
+					checked={ this.state.jetpack_firewall_ip_list }
+					disabled={
+						! isFirewallActive ||
+						unavailableInOfflineMode ||
+						this.props.isSavingAnyOption( [ 'firewall', 'jetpack_firewall_ip_list' ] )
+					}
+					onChange={ this.handleIpListToggleChange }
+				>
+					<span className="jp-form-toggle-explanation">
+						{ __( 'Allow / Block list - Block or allow a specific request IP', 'jetpack' ) }
+					</span>
+				</CompactFormToggle>
+
+				{ this.state.jetpack_firewall_ip_list && (
+					<>
+						<div className="firewall__settings__ips">
+							<FormLabel>{ __( 'Blocked IP addresses', 'jetpack' ) }</FormLabel>
+							<Textarea
+								disabled={
+									! isFirewallActive ||
+									unavailableInOfflineMode ||
+									this.props.isSavingAnyOption( [
+										'firewall',
+										'jetpack_firewall_ip_list',
+										'jetpack_firewall_ip_block_list',
+									] )
+								}
+								name="jetpack_firewall_ip_block_list"
+								placeholder={ __( 'Example: 12.12.12.1-12.12.12.100', 'jetpack' ) }
```

Comment:
as of now, the IP checks do not support ranges. We are expecting an array of single IP addresses, so it would probably make sense to reflect that somehow here (e.g. either 1 IP per line or comma separated). Ranges and CIDR blocks will certainly come in a future version.

Reply:
I've updated the placeholder here to display a list of single IP addresses :+1: